File: apps/frontend/lib/api/careerpilot.ts
import { apiFetch, apiPost } from '@/lib/api/client';

export interface CareerPilotJob {
  id: string;
  external_id?: string | null;
  provider?: string;
  company: string;
  title: string;
  description: string;
  location?: string | null;
  work_mode?: string | null;
  employment_type?: string | null;
  salary_min?: number | null;
  salary_max?: number | null;
  currency?: string | null;
  experience_required?: string | null;
  skills_required?: string[];
  skills_preferred?: string[];
  qualifications?: string[];
  hard_requirements?: string[];
  application_url?: string | null;
  posted_at?: string | null;
  discovered_at?: string | null;
}

export interface CareerPilotMatchResult {
  overall_score: number;
  recommendation: string;
  matched_requirements: string[];
  partially_matched_requirements: string[];
  missing_requirements: string[];
  hard_requirement_failures: string[];
  evidence: Array<{
    requirement: string;
    status: string;
    evidence?: string | null;
    source?: string;
  }>;
  score_breakdown: Record<string, number>;
}

export async function fetchCareerPilotJobs(): Promise<CareerPilotJob[]> {
  const response = await apiFetch('/careerpilot/jobs', { credentials: 'include' });
  if (!response.ok) {
    throw new Error('Unable to load recommended jobs.');
  }
  const payload = await response.json();
  return payload.jobs ?? [];
}

export async function discoverCareerPilotJobs(): Promise<CareerPilotJob[]> {
  const response = await apiPost('/careerpilot/jobs/discover', {
    criteria: {},
    persist: true,
  });
  if (!response.ok) {
    throw new Error('Error discovering jobs');
  }
  const payload = await response.json();
  return payload.jobs ?? [];
}

export async function fetchCareerPilotJob(jobId: string): Promise<CareerPilotJob> {
  const response = await apiFetch(`/careerpilot/jobs/${encodeURIComponent(jobId)}`, {
    credentials: 'include',
  });
  if (!response.ok) {
    throw new Error('Unable to load job details.');
  }
  const payload = await response.json();
  return payload.job;
}

export async function fetchCareerPilotMatch(jobId: string): Promise<CareerPilotMatchResult> {
  const response = await apiFetch(`/careerpilot/jobs/${encodeURIComponent(jobId)}/match`, {
    credentials: 'include',
  });
  if (!response.ok) {
    const payload = await response.json().catch(() => null);
    const detail = typeof payload?.detail === 'string' ? payload.detail : null;
    throw new Error(detail ?? 'Unable to load job match.');
  }
  const payload = await response.json();
  return payload.match;
}

File: apps/backend/app/schemas/careerpilot.py
"""Pydantic models for the CareerPilot vertical slice."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class CareerPilotJob(BaseModel):
    id: str
    external_id: str | None = None
    provider: str = "demo"
    company: str
    title: str
    description: str
    location: str | None = None
    work_mode: str | None = None
    employment_type: str | None = None
    salary_min: int | None = None
    salary_max: int | None = None
    currency: str | None = None
    experience_required: str | None = None
    skills_required: list[str] = Field(default_factory=list)
    skills_preferred: list[str] = Field(default_factory=list)
    qualifications: list[str] = Field(default_factory=list)
    hard_requirements: list[str] = Field(default_factory=list)
    application_url: str | None = None
    posted_at: str | None = None
    discovered_at: str | None = None
    provider_metadata: dict[str, Any] = Field(default_factory=dict)
    screening_questions: list[dict[str, Any]] = Field(default_factory=list)


class CareerPilotDiscoveryRequest(BaseModel):
    criteria: dict[str, Any] = Field(default_factory=dict)
    persist: bool = True


class CareerPilotDiscoveryResponse(BaseModel):
    jobs: list[CareerPilotJob]
    total: int
    demo_mode: bool = True


class CareerPilotMatchResult(BaseModel):
    overall_score: int
    recommendation: str
    matched_requirements: list[str] = Field(default_factory=list)
    partially_matched_requirements: list[str] = Field(default_factory=list)
    missing_requirements: list[str] = Field(default_factory=list)
    hard_requirement_failures: list[str] = Field(default_factory=list)
    evidence: list[dict[str, Any]] = Field(default_factory=list)
    score_breakdown: dict[str, int] = Field(default_factory=dict)


class CareerPilotJobMatchResponse(BaseModel):
    job: CareerPilotJob
    match: CareerPilotMatchResult


class CareerPilotJobDetailResponse(BaseModel):
    job: CareerPilotJob
